server/lib/pageExists.js

```diff
@@ -1,13 +1,21 @@
 const bluebird = require('bluebird');
 const query = require('./database').query;
 const escape = require('mysql').escape;
+const debug = require('debug')('perf-widget:lib:pageExists'); // eslint-disable-line no-unused-vars
 
-module.exports = bluebird.coroutine(function* pageExists(url) {
+const pageExists = bluebird.coroutine(function* pageExists(url) {
 	const command = `SELECT EXISTS(SELECT 1 FROM page WHERE url=${escape(url)})`
 
 	const result = yield query(command);
 
+	// Database closed the connection early return that the page does not exist
+	if (!result) {
+		return false;
+	}
+
 	const exists = result[0][Object.keys(result[0])[0]] === 1;
 
 	return exists;
 });
+
+module.exports = url => pageExists(url).catch(e => debug(e));
```

Comment:
The error looks to be handled by lines 11-15 in this file. Does the catch on this line (21) need to exist? Currently it would catch any errors thrown by pageExists, log them to stdout and then return a resolved promise with a value of `undefined`. Is this what we are wanting to do?